src/msg/async/ProtocolV2.cc

```diff
@@ -20,23 +21,57 @@ ostream &ProtocolV2::_conn_prefix(std::ostream *_dout) {
                 << " l=" << connection->policy.lossy << ").";
 }
 
-// TODO: REMOVE THIS
-void ProtocolV2::log(const std::string message, uint64_t val, uint64_t val2) {
-  ldout(cct, 1) << __func__ << " " << message << val << ":" << val2 << dendl;
-}
-
 using CtPtr = Ct<ProtocolV2> *;
 
+struct SHA256SignatureError : public std::exception {
+  std::string reason;
+
+  SHA256SignatureError(const char *sig1, const char *sig2) {
+    std::stringstream ss;
+
+    uint64_t s1 = *(uint64_t *)sig1;
+    uint64_t s2 = *(uint64_t *)(sig1 + 8);
+    uint64_t s3 = *(uint64_t *)(sig1 + 16);
+    uint64_t s4 = *(uint64_t *)(sig1 + 24);
+
+    uint64_t m1 = *(uint64_t *)sig2;
+    uint64_t m2 = *(uint64_t *)(sig2 + 8);
+    uint64_t m3 = *(uint64_t *)(sig2 + 16);
+    uint64_t m4 = *(uint64_t *)(sig2 + 24);
```

Comment:
let's use sha1_digest_t from include/types.h instead for this?  the above code is going to be inconsistent for big endian machines.

Reply:
done